src/thx/Iterables.hx

```diff
@@ -106,14 +106,14 @@ of its signature.
 /**
 Refer to `Array.map`.
 **/
-  inline public static function map<T, S>(it : Iterable<T>, f : T -> S) : Array<S>
-    return Iterators.map(it.iterator(), f);
+  inline public static function map<T, S>(it : Iterable<T>, f : T -> S) : Iterable<S>
```

Comment:
this and the following ones will break a lot of code. I am a little hesitant merging this.